loader.py

```diff
@@ -1,3 +1,182 @@
+import logging
+from os import walk
+from os.path import isdir, join, splitext
+from util import *
+
+logger = logging.getLogger(__name__)
+
+"""
+interface Organizer:
+
+    __init__(input_file=None)
+        Organizer can use labels from input_file
+        to create mappings into output_dir for
+        records.
+
+    get_dest_paths(record_files, output_dir)
+        Return dict of destination file locations
+        for record files in the output_dir.
+"""
+
+"""
+Mandatory keys in mapping dict.
+"""
+MAPPING_PROPS = {'old_path', 'new_path'}
+
+
+class DateBasedOrganizer:
+    """
+    Organize records (create mappings) based on
+    record creation time.
+    """
+
+    class Record:
+        """Util struct for working with record data."""
+        def __init__(self, filename, creation_time, duration):
+            self.filename = filename
+            self.creation_time = creation_time
+            self.duration = duration
+            self.finish_time = creation_time + duration
+
+        def overlap_with(self, record):
+            return ((self.creation_time < record.finish_time) and
+                    (record.creation_time < self.finish_time))
+
+    def __init__(self, input_file=None):
```

Comment:
toto by som sem nedaval, dokedy to nie je treba

Reply:
Zacal som to kodit ked som ratal so vstupnym fajlom a potom som to nechal aby to bolo kompatibilne s interfejsom, ale pravda, je to tam zbytocne.